b2sdk/file_version.py

```diff
@@ -300,10 +297,32 @@ def headers_size(self) -> int:
             file_retention=self.file_retention,
             legal_hold=self.legal_hold,
         )
-        headers_str = '\n'.join(
-            f'{key}: {value}' for key, value in headers.items() if value is not None
+
+        newline, separator = ('\n', ': ') if not omit_delimiters else ('', '')
+        headers_str = newline.join(
+            f'{key}{separator}{value}' for key, value in headers.items() if value is not None
         )
-        return len(headers_str.encode('utf8'))
+        return headers_str.encode('utf8')
+
+    @property
+    def has_large_header(self) -> bool:
+        """
+        Determine whether FileVersion's info fits header size limit defined by B2.
+
+        See https://www.backblaze.com/b2/docs/files.html#httpHeaderSizeLimit.
+        """
+
+        # For files encrypted with Server-Side Encryption and/or in
+        # File Lock-enabled buckets, the limit is reduced to 2,048 bytes
+        if self.server_side_encryption.mode == EncryptionMode.SSE_C or \
```

Comment:
actually it should be encryption mode other than EncryptionMode.NONE - EncryptionMode.SSE_B2 also has 2048b limit